core/src/gl/shaderProgram.cpp

```diff
@@ -284,13 +284,10 @@ std::string ShaderProgram::applySourceBlocks(const std::string& source, bool fra
 
     auto sourcePos = source.begin();
 
-    auto linesBegin = std::sregex_iterator(source.begin(), source.end(), pragmaLine);
-    auto linesEnd = std::sregex_iterator();
+    for (auto it = source.begin(); std::regex_search(it, source.end(), sm, pragmaLine);) {
+        it += sm.position() + sm.length();
```

Comment:
yeah, only std::regex_iterator seems to be broken a rather obscure feature. if regex_search gave the wrong position someone would probably have noticed before release :)